### PR TITLE
Animation_NodeMisc Model Group

### PR DESCRIPTION
**Pull request by stevk**
Further progress on #414 that covers edge cases

### CODE REVIEW

File: Source/FloatHelper.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace AssetGenerator
+{
+    static class FloatHelper
+    {
+        public static readonly float PiAsFloat = (float)Math.Round((decimal)(Math.PI));
```
Comment (bghgary) [inline Source/FloatHelper.cs]: Just put the value as a constant.

File: Source/FloatHelper.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+
+namespace AssetGenerator
+{
+    static class FloatHelper
```
Comment (bghgary) [inline Source/FloatHelper.cs]: I would name the class something like FloatMath (Unity calls it Mathf). The constant should just be `PI`. No need to call it PiAsFloat since it's already under a class that means float.

File: Source/FloatHelper.cs (hunk 1 above)
Comment (stevk) [reply Source/FloatHelper.cs]: To my understanding, I can't unless I just write out the exact value of pi (instead of using Math.PI). 

`The expression being assigned to 'FloatHelper.Pi' must be constant`

Comment (stevk) [reply Source/FloatHelper.cs]: Fixed

File: Source/FloatHelper.cs (hunk 2 above)
Comment (stevk) [reply Source/FloatHelper.cs]: Fixed

File: Source/FloatMath.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿namespace AssetGenerator
+{
+    static class FloatMath
+    {
+        public const float Pi = 3.14159274f;
```
Comment (bghgary) [inline Source/FloatMath.cs]: nit: 3.14159274 -> 3.14159265 so that it's more readable

File: Output/Animation_NodeMisc/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+These models are intended to test edge cases of an animation on a node.  
+
+The following table shows the properties that are set for a given model.  
+
+|   | Sample Image | Description |
+| :---: | :---: | :---: |
+| [00](Animation_NodeMisc_00.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=0) | [<img src="Figures/Thumbnails/Animation_NodeMisc_00.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_00.png) | Multiple channels are used, each with the same start and end time. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: nit: Should specify what the targets are.

File: Output/Animation_NodeMisc/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+| [02](Animation_NodeMisc_02.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=2) | [<img src="Figures/Thumbnails/Animation_NodeMisc_02.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_02.png) | There are two channels. The first channel has a constant transform value, starts after the second channel, and targets translation. The second channel targets rotation, has a first keyframe which starts above zero, and ends before the other channel. |
+| [03](Animation_NodeMisc_03.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=3) | [<img src="Figures/Thumbnails/Animation_NodeMisc_03.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_03.png) | The channel has only one keyframe and targets translation. |
+| [04](Animation_NodeMisc_04.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=4) | [<img src="Figures/Thumbnails/Animation_NodeMisc_04.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_04.png) | There are two channels, the first targeting rotation and the second scale. The rotation is applied to the left node, and the scale to the right node. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: nit: Run on sentence.

File: Source/ModelGroups/Animation_NodeMisc.cs [hunk 6]
```diff
@@ -320,8 +327,8 @@ void CreateMultipleChannelsForDifferentNodes(List<Runtime.AnimationChannel> chan
                     SetLinearSamplerForRotation(channels[0]);
                     SetTranslationChannelTarget(animations[1].Channels[0], nodes[0]);
                     SetLinearSamplerForTranslation(animations[1].Channels[0]);
-                    properties.Add(new Property(PropertyName.Description, "There are two animations. The first animation targets rotation, while the second targets rotation. " +
-                        "Neither animation interacts with the other or bleeds across."));
+                    properties.Add(new Property(PropertyName.Description,
+                        "There are two animations, each with one channel. The first animation's channel targets rotation. The second animation's channel targets translation. Neither animation interacts with the other or bleeds across."));
```
Comment (bghgary) [inline Source/ModelGroups/Animation_NodeMisc.cs]: Remove last statement.

File: Output/Animation_NodeMisc/README.md [hunk 7]
```diff
@@ -4,12 +4,12 @@ The following table shows the properties that are set for a given model.
...
-| [06](Animation_NodeMisc_06.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=6) | [<img src="Figures/Thumbnails/Animation_NodeMisc_06.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_06.png) | The model is rotated, and then an animation that targets translation with a constant value is applied. Neither of these override the other. |
-| [07](Animation_NodeMisc_07.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=7) | [<img src="Figures/Thumbnails/Animation_NodeMisc_07.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_07.png) | There are two animations. The first animation targets rotation, while the second targets rotation. Neither animation interacts with the other or bleeds across. |
+| [00](Animation_NodeMisc_00.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=0) | [<img src="Figures/Thumbnails/Animation_NodeMisc_00.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_00.png) | There are two channels. The first channel targets translation. The second channel targets rotation. The start and end times of both channels are `0.0` and `4.0` respectively. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: need nbsp for code spaces

Comment (stevk) [reply Output/Animation_NodeMisc/README.md]: Fixed

File: Source/ModelGroups/Animation_NodeMisc.cs (hunk 6 above)
Comment (stevk) [reply Source/ModelGroups/Animation_NodeMisc.cs]: Fixed

**Comment by stevk** [comment]
- [ ]  Change model 02 so that it isn't cut off by the side of the screen in the viewer
- [ ]  Make sure model 03 not rendering isn't a problem with the model itself
- [ ]  Change model 04 so that it looks less trippy

**Comment by stevk** [comment]
@bghgary and @kcoley The issue with model 03 looks to me like a bug in Babylon.js. The other model have been updated per your feedback.

**Comment by bghgary** [comment]
Confirmed. This is a bug.

**Comment by bghgary** [comment]
This is fixed now.

**Comment by stevk** [comment]
@kcoley Sorry to ask for another so soon, but could you update the screenshot generator to the latest version of babylon again?

**Comment by kcoley** [comment]
@stevk sure, I'll work on it now.

**Comment by kcoley** [comment]
@stevk looks like Babylon's npm package was not updated recently.  You will have to update the files manually for now.

File: Output/Animation_NodeMisc/README.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+| [01](Animation_NodeMisc_01.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=1) | [<img src="Figures/Thumbnails/Animation_NodeMisc_01.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_01.png) | There is one channel with a non-zero start time. The channel targets rotation. The start time is `1.0`. |
+| [02](Animation_NodeMisc_02.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=2) | [<img src="Figures/Thumbnails/Animation_NodeMisc_02.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_02.png) | There are two channels with different start and end times. The first channel targets translation with a constant value of <code>[0.0,&nbsp;0.1,&nbsp;0.0]</code> with start and end times of `2.0` and `6.0` respectively. The second channel targets rotation with start and end times of `1.0` and `5.0` respectively. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: Change the first channel to animate between -0.1 and 0.1 instead of constant.

File: Output/Animation_NodeMisc/README.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+| [03](Animation_NodeMisc_03.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=3) | [<img src="Figures/Thumbnails/Animation_NodeMisc_03.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_03.png) | There is one channel with only one keyframe. The channel targets translation with a value of <code>[-0.1,&nbsp;0.0,&nbsp;0.0]</code>. |
+| [04](Animation_NodeMisc_04.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=4) | [<img src="Figures/Thumbnails/Animation_NodeMisc_04.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_04.png) | There are two channels with different node and path targets. The first channel targets the left node and rotation. The second channel targets the right node and scale. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: Point of this test is to target different nodes. Change path to rotation and translation (up and down) instead.

File: Output/Animation_NodeMisc/README.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+| [05](Animation_NodeMisc_05.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=5) | [<img src="Figures/Thumbnails/Animation_NodeMisc_05.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_05.png) | There is one channel that targets a node. The node has a rotation of <code>[0.0,&nbsp;0.707,&nbsp;0.0,&nbsp;0.707]</code>. The channel overrides the rotation of the node to a different constant value of <code>[0.0,&nbsp;-0.707,&nbsp;0.0,&nbsp;-0.707]</code>. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: Use translation instead of rotation.

File: Output/Animation_NodeMisc/README.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+| [06](Animation_NodeMisc_06.gltf)<br>[View](https://bghgary.github.io/glTF-Assets-Viewer/?folder=1&model=6) | [<img src="Figures/Thumbnails/Animation_NodeMisc_06.png" align="middle">](Figures/SampleImages/Animation_NodeMisc_06.png) | There is one channel that targets a node. The node has a rotation of <code>[0.0,&nbsp;0.707,&nbsp;0.0,&nbsp;0.707]</code>. The channel targets the translation of the node with a constant value of <code>[0.0,&nbsp;0.1,&nbsp;0.0]</code>. |
```
Comment (bghgary) [inline Output/Animation_NodeMisc/README.md]: Use translation as the base and use non-constant animation for rotation.

**Comment by stevk** [comment]
I've made the requested changes. 
- 05 the base translation is unexpectedly kept in addition to the translation animation
- 06 the base translation is unexpectedly overridden by the rotation animation


I think these two things are causing issues with positioning on model 04 as well.

**Comment by stevk** [comment]
@bghgary I've fixed models 05 and 06. 

04 still isn't being framed correctly. One of the nodes is centered on 0,0,0 while the other is offset by -0.8, 0, 0. The camera is centered as expected, but the sides of both nodes are cutoff.

Readme : https://github.com/stevk/glTF-Asset-Generator/tree/nodeanimationextras/Output/Animation_NodeMisc

View Link: https://bghgary.github.io/glTF-Assets-Viewer/?manifest=https://raw.githubusercontent.com/stevk/glTF-Asset-Generator/nodeanimationextras/Output/Manifest.json&folder=1&model=4
